src/backend/metal/src/window.rs

```diff
@@ -405,6 +405,14 @@ impl Device {
         let cmd_queue = self.shared.queue.lock();
 
         unsafe {
+            // On iOS, unless the user supplies a view with a CAMetalLayer, we
+            // create one as a sublayer. However, when the view changes size,
+            // its sublayers are not automatically resized, and we must resize
+            // it here. The drawable size and the layer size don't correlate
+            let view = surface.inner.view.unwrap().as_ptr();
```

Comment:
should this be ios-only?

Reply:
It didn't seem to have any ill effect on macOS, so it probably doesn't matter. We could make it iOS only. On a separate note, it occurred to me that `view` may be `None` if the surface is created with `create_surface_from_layer` instead of `create_surface_from_uiview`. I'll address these now.